src/go/k8s/pkg/resources/secret.go

```diff
@@ -0,0 +1,247 @@
+	scheme *runtime.Scheme,
+	serviceFQDN string,
+	pandaproxySASLUser types.NamespacedName,
+	schemaRegistrySASLUser types.NamespacedName,
+	logger logr.Logger,
+) *SecretResource {
+	return &SecretResource{
+		client,
+		scheme,
+		pandaCluster,
+		serviceFQDN,
+		pandaproxySASLUser,
+		schemaRegistrySASLUser,
+		logger.WithValues("Kind", secretKind()),
+	}
+}
+
+// Ensure will manage kubernetes v1.Secret for redpanda STS
+func (r *SecretResource) Ensure(ctx context.Context) error {
+	obj, err := r.obj()
+	if err != nil {
+		return fmt.Errorf("unable to construct object: %w", err)
+	}
+	created, err := CreateIfNotExists(ctx, r, obj, r.logger)
+	if err != nil || created {
+		return err
+	}
+	var secret corev1.Secret
+	err = r.Get(ctx, r.Key(), &secret)
+	if err != nil {
+		return fmt.Errorf("error while fetching Secret resource: %w", err)
+	}
+
+	return r.update(ctx, &secret, obj.(*corev1.Secret), r.Client, r.logger)
+}
+
+func (r *SecretResource) update(
+	ctx context.Context,
+	current *corev1.Secret,
+	modified *corev1.Secret,
+	c k8sclient.Client,
+	logger logr.Logger,
+) error {
+	// Do not touch existing last-applied-configuration (it's not reconciled in the main loop)
+	if val, ok := current.Annotations[LastAppliedConfigurationAnnotationKey]; ok {
+		if modified.Annotations == nil {
+			modified.Annotations = make(map[string]string)
+		}
+		modified.Annotations[LastAppliedConfigurationAnnotationKey] = val
+	}
+
+	_, err := Update(ctx, current, modified, c, logger)
+	return err
+}
+
+// obj returns resource managed client.Object
+func (r *SecretResource) obj() (k8sclient.Object, error) {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.Key().Namespace,
+			Name:      r.Key().Name,
+			Labels:    labels.ForCluster(r.pandaCluster),
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		Data: map[string][]byte{},
+	}
+
+	secret.Data[postStartKey] = []byte(r.getPostStartScript())
+	secret.Data[preStopKey] = []byte(r.getPreStopScript())
+
+	err := controllerutil.SetControllerReference(r.pandaCluster, secret, r.scheme)
+	if err != nil {
+		return nil, err
+	}
+
+	return secret, nil
+}
+
+// Key returns namespace/name object that is used to identify object.
+// For reference please visit types.NamespacedName docs in k8s.io/apimachinery
+func (r *SecretResource) Key() types.NamespacedName {
+	return SecretKey(r.pandaCluster)
+}
+
+// SecretKey provides config map name that derived from redpanda.vectorized.io CR
+func SecretKey(pandaCluster *redpandav1alpha1.Cluster) types.NamespacedName {
+	return types.NamespacedName{Name: resourceNameTrim(pandaCluster.Name, baseSuffix), Namespace: pandaCluster.Namespace}
+}
+
+func secretKind() string {
+	var cfg corev1.Secret
+	return cfg.Kind
+}
+
+// getPostStartScript creates a script that removes maintenance mode after startup.
+func (r *SecretResource) getPostStartScript() string {
+	// TODO replace scripts with proper RPK calls
```

Comment:
🙏 that would be great, but I understand we're in time pressure

Reply:
Not only that, but this wasn't even my comment! :D